tests/tests_api/test_api_ext.py

```diff
@@ -72,3 +73,23 @@ def test_web_client_cannot_access_session_attribute():
     ) as web_client:
         with pytest.raises(AttributeError):
             assert web_client.headers
+
+
+@pytest.mark.parametrize(
```

Comment:
you could use [hypothesis](https://hypothesis.readthedocs.io) here

Reply:
But you cannot test the result is correctly formatted?
I used hypothesis for the `test_it_raise_error_while_format_headers_error` test